### PR TITLE
example: Grafbase ⨯ Clerk

### PR DESCRIPTION
**Pull request by notrab**
# Description

Adds a new example using Grafbase, Clerk, and Next.js!

Resolves GB-1809

# Type of change

- [ ] 💔 Breaking
- [x] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: examples/clerk-nextjs/.env.example [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Refer to the README on how to fill these variables
```
Comment (fbjork) [inline examples/clerk-nextjs/.env.example:1]: I would prefix by framework (nextjs-clerk). That way they are grouped together by framework.

File: examples/clerk-nextjs/.env.example [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+NEXT_PUBLIC_CLERK_FRONTEND_API=
+CLERK_API_KEY=
+NEXT_PUBLIC_GRAFBASE_URL=http://localhost:4000/graphql
```
Comment (fbjork) [inline examples/clerk-nextjs/.env.example:5]:
```suggestion
NEXT_PUBLIC_GRAFBASE_API_URL=http://localhost:4000/graphql
```

File: examples/clerk-nextjs/.env.example [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Refer to the README on how to fill these variables
+
+NEXT_PUBLIC_CLERK_FRONTEND_API=
+CLERK_API_KEY=
```
Comment (fbjork) [inline examples/clerk-nextjs/.env.example:4]: Is this needed?

File: examples/clerk-nextjs/package.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "clerk-grafbase-starter",
```
Comment (fbjork) [inline examples/clerk-nextjs/package.json:2]: `nextjs-clerk`

File: examples/clerk-nextjs/pages/sign-up/[[...index]].tsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { SignUp } from '@clerk/nextjs'
```
Comment (fbjork) [inline examples/clerk-nextjs/pages/sign-up/[[...index]].tsx:1]: Why isn't this just a page named sign-up.tsx?

Comment (notrab) [reply examples/clerk-nextjs/pages/sign-up/[[...index]].tsx:1]: I just copy/pasted from the `hasura` starter. I'll tidy these up later. Right now I've just made this work to start writing the guide.

File: examples/clerk-nextjs/pages/schema.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  const { getToken } = useAuth()
+
+  const fetcher = async () =>
+    await fetch(process.env.NEXT_PUBLIC_GRAFBASE_URL as string, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        authorization: `Bearer ${await getToken({
```
Comment (vitalikda) [inline examples/clerk-nextjs/pages/schema.tsx:16]:
```suggestion
        'x-api-key': `${await getToken({
```

Comment (notrab) [reply examples/clerk-nextjs/pages/schema.tsx:16]: Is this right? I was under the impression `x-api-key` was for API keys made on Grafbase?

File: examples/clerk-nextjs/grafbase/schema.graphql [hunk 7]
```diff
@@ -0,0 +1,19 @@
+schema
+  @auth(
+    providers: [{ type: oidc, issuer: "CLERK_ISSUER_URL" }]
```
Comment (DougAtGrafbase) [inline examples/clerk-nextjs/grafbase/schema.graphql:3]: Let's make this an environment variable once they are supported.

File: examples/clerk-nextjs/.env.example (hunk 3 above)
Comment (notrab) [reply examples/clerk-nextjs/.env.example]: Yeah. The Clerk provider needs it, and reads it using the same ENV name as explained in their docs.

**Comment by notrab** [comment]
@vitalikda @fbjork The `sign-in/[[...index]]` is actually a convention Clerk uses as it captures redirects, and special flows where the URL is `/sign-in/verify-email...` for example so they have to follow that convention.

I've updated the ENV name as well as renamed the folder to `nextjs-clerk`.

Let's switch to using environment variables from Grafbase for the issuer URL and document that in the readme too.

**Comment by notrab** [comment]
Maybe we disable actions for `examples/templates` folders?

**Comment by vitalikda** [comment]
> Maybe we disable actions for `examples/templates` folders?

we discussed it with @yoav-lavi before but looks like it's not possible

**Comment by yoav-lavi** [comment]
I didn't look into it for that long, we could revisit

**Review by vitalikda — APPROVED** [review]
LGTM

**Comment by yoav-lavi** [comment]
Rerunning tests after fixing the S3 permissions

**Pull request merged**